erpnext/selling/page/point_of_sale/point_of_sale.js

```diff
@@ -50,11 +50,11 @@ erpnext.pos.PointOfSale = class PointOfSale {
 				this.set_online_status();
 			},
 			() => this.setup_company(),
-
 			() => this.make_new_invoice(),
 			() => {
 				frappe.dom.unfreeze();
 			},
+			() => this.fetch_pos_profile_data(),
```

Comment:
No need to write extra method, use existing method.
Set value in the method set_pos_fields, you can add the field allow_print_before_pay in the sales invoice and use same in the POS.
https://github.com/frappe/erpnext/blob/develop/erpnext/accounts/doctype/sales_invoice/sales_invoice.py#L293